utils/media.js

```diff
@@ -0,0 +1,87 @@
+const multer = require("multer");
+const path = require("path");
+const fse = require("fs-extra");
+const get = require("lodash/get");
+
+function createMedia(root) {
+  const storage = multer.diskStorage({
+    async destination(req, file, callback) {
+      let destinationDir = path.resolve(
+        root,
+        process.env.MEDIA_UPLOAD_DIRECTORY,
+      );
+      if (process.env.MEDIA_DATES_FOLDER) {
+        destinationDir = path.resolve(
+          destinationDir,
+          String(new Date().getFullYear()),
+          String(new Date().getMonth() + 1),
+          String(new Date().getDate()),
+        );
+      }
+      await fse.ensureDir(destinationDir);
+
+      callback(null, destinationDir);
+    },
+    async filename(req, file, cb) {
+      const uniqueSuffix = `${Date.now()}-${Math.round(Math.random() * 1e9)}`;
+      const fileMeta = path.parse(file.originalname);
+      cb(null, `${fileMeta.name}-${uniqueSuffix}${fileMeta.ext}`);
+    },
+  });
+  const upload = multer({ storage });
+
+  const appendSrc = (file) => {
+    return {
+      ...file,
+      src: `${process.env.MEDIA_PUBLIC_BASE_URL}${file.relativeSrc}`,
```

Comment:
`process.env.MEDIA_PUBLIC_BASE_URL` can be removed as we discussed about moving to client layer